File: Examples/Java/SDK/src/main/java/com/aspose/imaging/cloud/examples/images/UpdateBmpProperties.java
package com.aspose.imaging.cloud.examples.images;

import com.aspose.imaging.cloud.examples.Utils;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardCopyOption;

public class UpdateBmpProperties {

    public static void main(String... args) throws IOException {
        String input = "Sample1.bmp";
        String output = "Sample2.bmp";
        Path inputFile = Utils.getPath(UpdateBmpProperties.class, input);
        Path outputFile = Utils.getPath(UpdateBmpProperties.class, output);

        Integer bitsPerPixel = 24;
        Integer horizontalResolution = 300;
        Integer verticalResolution = 300;
        Boolean fromScratch = false;
        String outPath = "";

        Utils.getStorageSdk().PutCreate(
                input,
                null,
                Utils.STORAGE,
                inputFile.toFile()
        );

        Utils.getImagingSdk().GetImageBmp(
                input,
                bitsPerPixel,
                horizontalResolution,
                verticalResolution,
                fromScratch,
                outPath,
                Utils.FOLDER,
                Utils.STORAGE
        );

        com.aspose.storage.model.ResponseMessage sr
                = Utils.getStorageSdk().GetDownload(
                        input,
                        null,
                        Utils.STORAGE
                );

        Files.copy(sr.getInputStream(), outputFile, StandardCopyOption.REPLACE_EXISTING);

    }
}

File: Examples/DotNET/SDK/VisualBasic/Images/UpdateBmpProperties.vb
'////////////////////////////////////////////////////////////////////////
' Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
'
' This file is part of Aspose.Imaging. The source code in this file
' is only intended as a supplement to the documentation, and is provided
' "as is", without warranty of any kind, either expressed or implied.
'////////////////////////////////////////////////////////////////////////

Imports System
Imports Aspose.Cloud

Namespace Aspose.Imaging.Cloud.Examples.Images
	Friend Class UpdateBmpProperties
		Shared Sub Main()
			Dim dataDir As String = Aspose.Imaging.Cloud.Examples.Common.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)
			Dim input As String = "sample1.bmp"
			Dim output As String = "output.bmp"

			Common.StorageService.File.UploadFile(dataDir & input, input, storage:= Common.STORAGE)

			Dim bitsPerPixel As Integer = 16
			Dim horizontalResolution As Integer = 1240
			Dim verticalResolution As Integer = 1240
			Dim outPath As String = "Imaging/" & input

			Common.GetImagingSdk().Bmp.UpdateParametersOfBmpImage(input, bitsPerPixel, horizontalResolution, verticalResolution, outPath, Common.FOLDER, storage:= Common.STORAGE)

            Common.StorageService.File.DownloadFile(outPath, dataDir & output, storage:=Common.STORAGE)

        End Sub
	End Class
End Namespace
